ngraph_bridge/ngraph_builder.cc

```diff
@@ -4825,6 +4825,55 @@ static Status TranslateUnpackOp(const Node* op,
   return Status::OK();
 }
 
+static Status TranslateUnsortedSegmentSumOp(
+    const Node* op, const std::vector<const Tensor*>& static_input_map,
+    Builder::OpMap& ng_op_map) {
+  /*
+  # Code from python that reflects how unsorted_segment_sum
+  # works in tensorflow using scatter_add
+
+  def unsorted_segment_sum(inputs, segment_ids, num_segments):
+    inputs = tf.convert_to_tensor(inputs)
+    segment_ids = tf.convert_to_tensor(segment_ids)
+
+    shape = inputs.get_shape().as_list()
+    seg_ids = segment_ids.get_shape().as_list()
+    result = tf.Variable(tf.zeros([num_segments] + shape[len(seg_ids):],
+                                  dtype=inputs.dtype))
+
+    return tf.scatter_add(result, segment_ids, inputs)
+  */
+
+  shared_ptr<ng::Node> ng_input, ng_segment_ids, ng_num_segments;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input, &ng_segment_ids,
+                                   &ng_num_segments));
+
+  int num_segments;
+  std::vector<int64> tmp_num_segments;
+  TF_RETURN_IF_ERROR(
+      GetStaticInputVector(op, 2, static_input_map, &tmp_num_segments));
+  num_segments = tmp_num_segments[0];
```

Comment:
Do we need to check and return a bad status if `num_segments<max(ng_num_segments)`.. or maybe TF itself throws an error if we try to do that?

Reply:
Should we add a test for the same like a test for the failing case?

Reply:
Can you point me to any example of C++ test in bridge that tests for failure?